addon/components/pie-chart.js

```diff
@@ -483,9 +483,17 @@ export default ChartComponent.extend(FloatingTooltipMixin,
     groups.select('path').attr(this.get('sliceAttrs'));
     
     var labelWidth = this.get('labelWidth');
+    var maxLabelWidth = this.get('outerWidth')/2 - this.get('labelPadding');
     var labelTrimmer = LabelTrimmer.create({
-      getLabelSize: function() {
-        return labelWidth;
+      getLabelSize: function(d, selection) {
+        var text, xPos;
+        // Pick the current text element by filtering on data
+        text = selection.filter(function(data) {
+          return data === d;
+        });
+        // Calculate current horizontal translation from the center
+        xPos = d3.transform(text.attr("transform")).translate[0];
```

Comment:
+1 to Thang's comment. I don't really understand what this line does. Perhaps breaking it into multiple steps, and expanding on the comment would clarify